### PR TITLE
Tidy trailing commas

### PR DESCRIPTION
**Pull request by huong-li-nguyen**
## Description
See following discussion thread: https://github.com/mckinsey/vizro/pull/283#discussion_r1467341688

Idea was to tidy some of our code and remove redundant wrapping into multiple lines

If you see some spots where you would prefer multi-lines instead of a single line, you just have to add the trailing comma back. I've quickly scanned through the files and added some of the trailing commas (mostly for our example configurations).

**Note: Will be merged after https://github.com/mckinsey/vizro/pull/293**

## Notice

- [x] I acknowledge and agree that, by checking this box and clicking "Submit Pull Request":

  - I submit this contribution under the [Apache 2.0 license](https://www.apache.org/licenses/LICENSE-2.0.txt) and represent that I am entitled to do so on behalf of myself, my employer, or relevant third parties, as applicable.
  - I certify that (a) this contribution is my original creation and / or (b) to the extent it is not my original creation, I am authorized to submit this contribution on behalf of the original creator(s) or their licensees.
  - I certify that the use of this contribution as authorized by the Apache 2.0 license does not violate the intellectual property rights of anyone else.
  - I have not referenced individuals, products or companies in any commits, directly or indirectly.
  - I have not added data or restricted code in any commits, directly or indirectly.

### CODE REVIEW

File: vizro-ai/src/vizro_ai/chains/_llm_models.py [hunk 1]
```diff
@@ -12,26 +12,10 @@
...
-    },
-    {
-        "name": "gpt-4-1106-preview",
-        "max_tokens": 128000,
-        "wrapper": ChatOpenAI,
-    },
+    {"name": "gpt-3.5-turbo-0613", "max_tokens": 4096, "wrapper": ChatOpenAI},
+    {"name": "gpt-4-0613", "max_tokens": 8192, "wrapper": ChatOpenAI},
+    {"name": "gpt-3.5-turbo-1106", "max_tokens": 16385, "wrapper": ChatOpenAI},
+    {"name": "gpt-4-1106-preview", "max_tokens": 128000, "wrapper": ChatOpenAI},
```
Comment (petar-qb) [inline vizro-ai/src/vizro_ai/chains/_llm_models.py:18]: For this I would either undo the changes or create a separate `config` file where it would keep the `PREDEFINED_MODELS` in the expanded format.

File: vizro-core/src/vizro/actions/_action_loop/_build_action_loop_callbacks.py [hunk 2]
```diff
@@ -48,10 +48,7 @@ def _build_action_loop_callbacks() -> None:
         )
 
         gateway_inputs.append(
-            Input(
-                component_id={"type": "gateway_input", "trigger_id": actions_chain.id},
-                component_property="data",
-            )
+            Input(component_id={"type": "gateway_input", "trigger_id": actions_chain.id}, component_property="data")
```
Comment (petar-qb) [inline vizro-core/src/vizro/actions/_action_loop/_build_action_loop_callbacks.py:51]: This also looks more readable to me in the original format.

File: vizro-core/src/vizro/actions/_action_loop/_get_action_loop_components.py [hunk 3]
```diff
@@ -33,10 +33,7 @@ def _get_action_loop_components() -> html.Div:
     # Required to avoid the "Unknown callback Input" issue for multiple page app examples.
     components.extend(
         [
-            dcc.Store(
-                id={"type": "gateway_input", "trigger_id": actions_chain.id},
-                data=f"{actions_chain.id}",
-            )
+            dcc.Store(id={"type": "gateway_input", "trigger_id": actions_chain.id}, data=f"{actions_chain.id}")
```
Comment (petar-qb) [inline vizro-core/src/vizro/actions/_action_loop/_get_action_loop_components.py:36]: This also looks more readable to me in the original format.

File: vizro-core/src/vizro/models/_action/_action.py [hunk 4]
```diff
@@ -159,9 +153,7 @@ def build(self):
             "external": external_callback_inputs,
             "internal": {"trigger": Input({"type": "action_trigger", "action_name": self.id}, "data")},
         }
-        callback_outputs = {
-            "internal": {"action_finished": Output("action_finished", "data", allow_duplicate=True)},
-        }
+        callback_outputs = {"internal": {"action_finished": Output("action_finished", "data", allow_duplicate=True)}}
```
Comment (petar-qb) [inline vizro-core/src/vizro/models/_action/_action.py:156]: This also looks more readable to me in the original format.

File: vizro-core/src/vizro/models/_action/_action.py [hunk 5]
```diff
@@ -98,15 +98,9 @@ def _get_callback_mapping(self):
         return callback_inputs, callback_outputs, action_components
 
     def _action_callback_function(
-        self,
-        inputs: Union[Dict[str, Any], List[Any]],
-        outputs: Union[Dict[str, Output], List[Output], Output, None],
+        self, inputs: Union[Dict[str, Any], List[Any]], outputs: Union[Dict[str, Output], List[Output], Output, None]
```
Comment (petar-qb) [inline vizro-core/src/vizro/models/_action/_action.py:101]: This also looks more readable to me in the original format.

File: vizro-core/src/vizro/models/_components/form/range_slider.py [hunk 6]
```diff
@@ -83,14 +83,7 @@ def build(self):
...
             [
-                dcc.Store(
-                    f"{self.id}_callback_data",
-                    data={
-                        "id": self.id,
-                        "min": self.min,
-                        "max": self.max,
-                    },
-                ),
+                dcc.Store(f"{self.id}_callback_data", data={"id": self.id, "min": self.min, "max": self.max}),
```
Comment (petar-qb) [inline vizro-core/src/vizro/models/_components/form/range_slider.py:86]: Can we add `id` key for the first attribute? So:
`dcc.Store(id=f"{self.id}_callback_data", data={"id": self.id, "min": self.min, "max": self.max}),`

File: vizro-core/src/vizro/models/_components/form/slider.py [hunk 7]
```diff
@@ -79,14 +79,7 @@ def build(self):
...
             [
-                dcc.Store(
-                    f"{self.id}_callback_data",
-                    data={
-                        "id": self.id,
-                        "min": self.min,
-                        "max": self.max,
-                    },
-                ),
+                dcc.Store(f"{self.id}_callback_data", data={"id": self.id, "min": self.min, "max": self.max}),
```
Comment (petar-qb) [inline vizro-core/src/vizro/models/_components/form/slider.py:82]: I would add `id` here as well.

File: vizro-core/tests/unit/vizro/models/_components/form/test_range_slider.py [hunk 8]
```diff
@@ -17,14 +17,7 @@
...
         [
-            dcc.Store(
-                "range_slider_callback_data",
-                data={
-                    "id": "range_slider",
-                    "min": None,
-                    "max": None,
-                },
-            ),
+            dcc.Store("range_slider_callback_data", data={"id": "range_slider", "min": None, "max": None}),
```
Comment (petar-qb) [inline vizro-core/tests/unit/vizro/models/_components/form/test_range_slider.py:20]: I would also add `id` here.

File: vizro-core/tests/unit/vizro/models/_components/form/test_slider.py [hunk 9]
```diff
@@ -17,14 +17,7 @@
...
         [
-            dcc.Store(
-                "slider_id_callback_data",
-                data={
-                    "id": "slider_id",
-                    "min": 0,
-                    "max": 10,
-                },
-            ),
+            dcc.Store("slider_id_callback_data", data={"id": "slider_id", "min": 0, "max": 10}),
```
Comment (petar-qb) [inline vizro-core/tests/unit/vizro/models/_components/form/test_slider.py:20]: I would also add `id` here

File: vizro-core/tests/unit/vizro/models/_components/test_container.py [hunk 10]
```diff
@@ -23,10 +23,7 @@ def test_create_container_mandatory_only(self):
 
     def test_create_container_mandatory_and_optional(self):
         container = vm.Container(
-            title="Title",
-            components=[vm.Button(), vm.Button()],
-            id="my-id",
-            layout=vm.Layout(grid=[[0, 1]]),
+            title="Title", components=[vm.Button(), vm.Button()], id="my-id", layout=vm.Layout(grid=[[0, 1]])
```
Comment (petar-qb) [inline vizro-core/tests/unit/vizro/models/_components/test_container.py:26]: I'm always a fan of setting the `id` property as the first attribute, then `children` (or `components`), then the rest. It probably doesn't need to be part of this PR.

File: vizro-core/tests/unit/vizro/actions/_action_loop/test_get_action_loop_components.py [hunk 11]
```diff
@@ -118,15 +105,7 @@ def managers_one_page_two_components_two_controls(vizro_app, dash_data_table_wit
...
-    vm.Dashboard(
-        pages=[
-            vm.Page(
-                id="test_page_no_actions",
-                title="Second page",
-                components=[vm.Card(text="")],
-            )
-        ]
-    )
+    vm.Dashboard(pages=[vm.Page(id="test_page_no_actions", title="Second page", components=[vm.Card(text="")])])
```
Comment (petar-qb) [inline vizro-core/tests/unit/vizro/actions/_action_loop/test_get_action_loop_components.py:108]: This also looks more readable to me in the original format.

In general, I like condensing code in tests more than in src code, because the tests are more likely to be extended (by adding new tests) than to be changed or need to be understood in detail. However, this fixture is probably something that someone will have to understand.

**Review by petar-qb — APPROVED** [review]
First of all, thanks for taking care of this!
  
By my opinion, we should keep expanded format for code that is more likely to be changed or is more likely that someone will need to understand it in detail, otherwise I'd go with the condensed format.

I left comments in few places where I think we should keep older version of format.

**Review by nadijagraca — APPROVED** [review]
Looks good. 👍

**Review by antonymilne — APPROVED** [review]

File: vizro-ai/src/vizro_ai/chains/_llm_models.py (hunk 1 above)
Comment (huong-li-nguyen) [reply vizro-ai/src/vizro_ai/chains/_llm_models.py:18]: Yeah, no worries! I can just add the comma back. I didn't go through all of the hundreds of changes 😄 So there might be even some cases where we want to change it after merging this PR. In this case, we just have to add the comma back. Then it will be formatted correctly 👍

Comment (petar-qb) [reply vizro-ai/src/vizro_ai/chains/_llm_models.py:18]: Yeah, I know that all changes in this PR are caused as a result of adding or removing trailing commas. I just wanted to say that in this case, I would probably revert back the trailing comma after `"wrapper": ChatOpenAI` lines and keep original extended format.

File: vizro-core/tests/unit/vizro/models/_components/test_container.py (hunk 10 above)
Comment (huong-li-nguyen) [reply vizro-core/tests/unit/vizro/models/_components/test_container.py:26]: I don't think we need to be super strict about this, as position shouldn't matter given that we use the keyword arguments. I can create a ticket though and we can consolidate this in another PR.

File: vizro-core/tests/unit/vizro/models/_components/form/test_slider.py (hunk 9 above)
Comment (huong-li-nguyen) [reply vizro-core/tests/unit/vizro/models/_components/form/test_slider.py:20]: I would create a separate ticket for this to go through all the tests and double-check where this doesn't apply yet. But I wouldn't do it in this PR, as this PR should only tidy the trailing commas that were added by black automatically.

Comment (petar-qb) [reply vizro-core/tests/unit/vizro/models/_components/form/test_slider.py:19]: Yeah sure. If you could just mention this PR link in that ticket (for the purpose of reminding) it would be awesome.

Comment (huong-li-nguyen) [reply vizro-core/tests/unit/vizro/models/_components/form/test_slider.py:19]: Already did :) https://github.com/orgs/McK-Internal/projects/15/views/12?pane=issue&itemId=51469120
